### DISCUSSION TITLE
Unstable Connection and High Amount of Packet Lost RTL8812AU and RPi4

### DISCUSSION THREAD

**Discussion by bugraaydogar** [svpcom/wfb-ng#145] [Questions] [answered]
Hi folks,

I recently started to explore low-latency video streaming domain. As a first step, I discovered openHD project and tested it with my setup which are 2 x RPi4 (8GB) + 2 x Tenda U12 Wifi Cards. Since openHD leverages the wifibroadcast, I wanted to dive into the wifibroadcast project. The problem I'm facing is that initially ~10-30 seconds, the video streaming is just perfect and then for some reasons, it starts to freeze and then lost completely. If I restart the wifibroadcast@gs service again, the connection establishes perfectly for ~10-30 secs then lost again. I can also verify that there is a huge packets loss.

I just could not figure out what is the problem. I have tested different channel from 1-165 but always the same behaviour. I also have seen that several people reported similar problem but seems like none of the discussions provides a solution.

Here is my environment details:

HW: RPi4 with 8GB
OS: Latest Raspbian OS (32 bit) => https://downloads.raspberrypi.org/raspios_full_armhf/images/raspios_full_armhf-2021-05-28/2021-05-07-raspios-buster-armhf-full.zip

RTL Drivers:  Manually compiled => https://github.com/svpcom/rtl8812au
Kernel Version: 5.10.17-v7l+

Configuration Details:

Grand station side;

```
[common]
wifi_channel = 165
wifi_region = 'BO'
wifi_txpower = 58

[gs_mavlink]
peer = 'connect://127.0.0.1:14550'

[gs_video]
peer = 'connect://127.0.0.1:5600'
bandwidth = 40 # I also tested with default 20 Mhz => no improvement
````

Drone side;

```
[common]
wifi_channel = 165
wifi_region = 'BO'
wifi_txpower = 58

[drone_mavlink]
peer = 'listen://127.0.0.1:14550'

[drone_video]
peer = 'listem://127.0.0.1:5600'
bandwidth = 40 # I also tested with default 20 Mhz => no improvement
````

I checked both wlan1 interfaces in both side to verify that they are using rtl88XXau driver by using `ethtool -i wlan1` command.
In addition to that, I can see that I have `drone-wfb` and `gs-wfb` but could ping each other. Finally, there is no networkmanager in the Raspian OS, therefore I disabled wlan0 by using `sudo ifconfig wlan0 down`

Any idea to help me proceed forward? I basically don't understand why I encounter with such issues since did not see any trouble with OpenHD.

Thanks
Bugra

**Comment by svpcom** [comment, marked as answer]
1. Check that there are no system processes (like NetworkManager, dhcpd, etc) which try to manage WFB network adapters
2. Check that your video bandwith < 0.5 of radio bandwidth (for example for MCS#1 max video bandwidth is 7mbit/s)

**Reply by svpcom** [reply]
```
[drone_video]
peer = 'listem://127.0.0.1:5600'
bandwidth = 40 # I also tested with default 20 Mhz => no improvement
```
you have typo in `listem:`. It should be `listen:`

**Reply by bugraaydogar** [reply]
Thanks @svpcom !
As you suggested, I disabled the dhcpcd.service by using `sudo systemctl disable dhcpcd.service`.
Also I added `denyinterfaces wlan0` to the file under /etc/dhcpcd.conf

My current observation is that there is no unstable connection anymore. However, from time to time (rarely), I have seen that there are some freezes around ~1-3 secs. Besides this, everything seems fine.

Would you mind explaining the relationship between disabling network-manager/dhcp and streaming the video from a wireless adaptor? Why it effects significantly? (I'm a bit newbie in the wireless  domain :) )

Thanks,
Bugra

**Reply by svpcom** [reply]
WFB uses wifi adapter in raw (monitor) mode. All generic tools (like NetworkManager, wpa_supplicant, dhcp, etc) try to switch it to normal mode.

**Reply by bugraaydogar** [reply]
Thanks.
